arm/entrypoint.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+set -e
+
+# Create Resource group if it does not exists
+
+RESOURCE_GROUP_EXISTS=$(az group exists -n "$AZURE_RESOURCE_GROUP")
+if [[ $RESOURCE_GROUP_EXISTS == "false" ]]
+then
+  if [[ -z "$RESOURCE_GROUP_LOCATION" ]]
+  then
+    echo "RESOURCE_GROUP_LOCATION is not set."
+  else
+    az group create --name "$AZURE_RESOURCE_GROUP" --location "$RESOURCE_GROUP_LOCATION"
+  fi
+fi
+
+URI_REGEX="^(http://|https://)\\w+"
+GUID=$(uuidgen | cut -d '-' -f 1)
+
+# Download parameters file if it is a remote URL
+
+if [[ $AZURE_TEMPLATE_PARAM_FILE =~ $URI_REGEX ]]
+then
+  PARAMETERS_FILE="parameters-${GUID}.json"
+  TEMPLATE_PARAMETERS=$(curl $AZURE_TEMPLATE_PARAM_FILE)
+  echo ${TEMPLATE_PARAMETERS} >> $PARAMETERS_FILE
```

Comment:
Do you need to create the file? Why not just have it in memory?

Reply:
Because if it is a file, the command format is --parameters @param_file_name , for inline it is --parameters 'params'. However i can handle this if condition. will do.